src/main/java/Ship/ShipManager.java

```diff
@@ -40,18 +41,36 @@ public void Draw(Graphics g) {
 		g.fillPolygon(myShip.getShape());
 	}
 	
+	
+	//needed to get ship health for the top bar
+	public int getShipHealth() {
+		return myShip.health;
+	}
+	
 	// collision detector
-	public ArrayList<Asteroid> detectCollisions(ArrayList<Asteroid> asteroids){
-		ArrayList<Asteroid> collisions = new ArrayList<Asteroid>();
+	public boolean detectCollisions(ArrayList<Asteroid> asteroids){
 		for (Asteroid a : asteroids) {
 			Area asteroidArea = new Area(a.getShape());
 			Area intersectionArea = new Area(myShip.getShape());
 			intersectionArea.intersect(asteroidArea);
 			
+			//collision detected
 			if (!intersectionArea.isEmpty()) {
-				collisions.add(a);
+			
+				//how much damage each asteroid does
+				myShip.decreaseHealth(5);
```

Comment:
With the total health being 10, this will destroy the ship in 2 hits. A suggestion for next step would be to set damage done by either how big the asteroid is or how big the collision is. I suggest doing it just based in size of the asteroid